src/qt/mAPreferencesWindow.cpp

```diff
@@ -116,67 +185,77 @@ void mAPreferencesWindow::configureDefaults()
-    QDialog(parent),
-    ui(new Ui::mAPreferencesWindow),
-    m_ma(ma)
-{
-    ui->setupUi(this);
-    
-    setWindowFlags(Qt::Window | Qt::WindowTitleHint | Qt::CustomizeWindowHint | Qt::WindowSystemMenuHint);
-    
-    m_colorDialog = NULL;
-    
-    m_indexToLabel.push_back("Normal Text"); m_indexToPref.push_back(mAPreferencesSyntaxColoringNormalText);
-    m_indexToLabel.push_back("Keywords");    m_indexToPref.push_back(mAPreferencesSyntaxColoringKeywords);
-    m_indexToLabel.push_back("Classes");     m_indexToPref.push_back(mAPreferencesSyntaxColoringClasses);
-    m_indexToLabel.push_back("UGens");       m_indexToPref.push_back(mAPreferencesSyntaxColoringUGens);
-    m_indexToLabel.push_back("Comments");    m_indexToPref.push_back(mAPreferencesSyntaxColoringComments);
-    m_indexToLabel.push_back("Strings");     m_indexToPref.push_back(mAPreferencesSyntaxColoringStrings);
-    m_indexToLabel.push_back("Numbers");     m_indexToPref.push_back(mAPreferencesSyntaxColoringNumbers);
-    m_indexToLabel.push_back("Background");  m_indexToPref.push_back(mAPreferencesSyntaxColoringBackground);
-    m_indexToColor.resize(m_indexToLabel.size());
-    
-    for(int i = 0; i < m_indexToLabel.size(); i++)
-        ui->syntaxColoringType->addItem(m_indexToLabel[i]);
-    
-    loadSettingsToGUI();
-    
-    ui->tabWidget->setCurrentIndex(0);    
-}
-
-mAPreferencesWindow::~mAPreferencesWindow()
-{
-    delete ui;
-}
 
+//-----------------------------------------------------------------------------
+// name: loadSettingsToGUI()
+// desc: load settings to GUI
+//-----------------------------------------------------------------------------
 void mAPreferencesWindow::loadSettingsToGUI()
 {
     ZSettings settings;
-    
+    // enable audio
     ui->enableAudio->setChecked(settings.get(mAPreferencesEnableAudio).toBool());
+    // accept network VM commands
     ui->enableNetworkVM->setChecked(settings.get(mAPreferencesEnableNetwork).toBool());
-    
+    // selecte buffer size
     ui->bufferSize->setCurrentIndex(ui->bufferSize->findText(QString("%1").arg(settings.get(mAPreferencesBufferSize).toInt())));
     
+    // audio driver selection | 1.5.0.1 (ge) added
+    char buffer[128];
+    // get driver from settings in the form of RtAudio::Api
+    int driver = settings.get(mAPreferencesAudioDriver).toInt();
+    // index of selected driver in the combobox
+    int selectedDriverIndex = -1;
+    // populate the "Audio drivers" ComboBox
+    for( unsigned int i = 0; i < rtaudio_num_compiled_apis; i++ )
+    {
+        // get full driver name
+        snprintf( buffer, sizeof(buffer), "%s", rtaudio_api_names[rtaudio_compiled_apis[i]][1] );
+        // add the item, with the Api enum as the associated data
+        ui->audioDriver->addItem( buffer, rtaudio_compiled_apis[i] );
+        // check if match with driver read in from settings
+        if( driver == rtaudio_compiled_apis[i] ) selectedDriverIndex = i;
+    }
```

Comment:
can try something like

```
    struct ChuckAudio::ApiInfo
    {
        t_CKUINT driver;
        std::string name;
        std::string userFriendlyName;
    };

    for( int i = 0; i < ChuckAudio::getNumDrivers(); i++ )
    {
        ChuckAudio::ApiInfo compiledApis = ChuckAudio::getCompiledDriverInfo(i);
        // get full driver name
        // add the item, with the Api enum as the associated data
        ui->audioDriver->addItem( api.userFriendlyName, api.driver );
        // check if match with driver read in from settings
        if( driver == compiledApis.driver ) selectedDriverIndex = i;
    }
```

Reply:
moved direct references to RtAudio out of miniAudicle preferences and accessed through ChuckAudio